include/chaiscript/language/chaiscript_engine.hpp

```diff
@@ -206,16 +206,16 @@ namespace chaiscript
 
     /// Skip BOM at the beginning of file
     static bool skip_bom(std::ifstream &infile) {
-        size_t bytes_needed = 3;
-        std::vector<char> v(bytes_needed);
+        std::streamsize bytes_needed = 3;
+        std::streamsize bytes_read = 0;
+        char buffer[3] = { '\0' };
```

Comment:
You don't have to initialize the array elements to `\0` if you check length anyway. This was more an idea in case you just do the three comparisons only and skip the length check.

But besides that, nothing more to comment on. :)